index.d.ts

```diff
@@ -0,0 +1,1353 @@
+
+    /**
+     * Enable Bluetooth. This function blocks until BLE is in PoweredOn state. [Android only]
+     *
+     * @param {?string} transactionId Transaction handle used to cancel operation
+     * @returns {Promise<BleManager>} Promise completes when state transition was successful.
+     */
+    enable(transactionId?: string): Promise<BleManager>;
+
+    /**
+     * Disable Bluetooth. This function blocks until BLE is in PoweredOff state. [Android only]
+     *
+     * @param {?string} transactionId Transaction handle used to cancel operation
+     * @returns {Promise<BleManager>} Promise completes when state transition was successful.
+     */
+    disable(transactionId?: string): Promise<BleManager>;
+
+    /**
+     * Current, global {@link State} of a {@link BleManager}. All APIs are working only when active state
+     * is "PoweredOn".
+     *
+     * @returns {Promise<State>} Promise which emits current state of BleManager.
+     */
+    state(): Promise<State> {
+      return this._callPromise(BleModule.state());
+    }
+
+    /**
+     * Notifies about {@link State} changes of a {@link BleManager}.
+     *
+     * @example
+     * const subscription = this.manager.onStateChange((state) => {
+     *      if (state === 'PoweredOn') {
+     *          this.scanAndConnect();
+     *          subscription.remove();
+     *      }
+     *  }, true);
+     *
+     * @param {function(newState: State)} listener Callback which emits state changes of BLE Manager.
+     * Look at {@link State} for possible values.
+     * @param {boolean} [emitCurrentState=false] If true, current state will be emitted as well. Defaults to false.
+     *
+     * @returns {Subscription} Subscription on which `remove()` function can be called to unsubscribe.
+     */
+    onStateChange(listener: (newState: State) => void, emitCurrentState: boolean = false): Subscription;
+
+    // Mark: Scanning ----------------------------------------------------------------------------------------------------
+
+    /**
+     * Starts device scanning. When previous scan is in progress it will be stopped before executing this command.
+     *
+     * @param {?Array<UUID>} UUIDs Array of strings containing {@link UUID}s of {@link Service}s which are registered in
+     * scanned {@link Device}. If `null` is passed, all available {@link Device}s will be scanned.
+     * @param {?ScanOptions} options Optional configuration for scanning operation.
+     * @param {function(error?: BleError, scannedDevice: ?Device)} listener Function which will be called for every scanned
+     * {@link Device} (devices may be scanned multiple times). It's first argument is potential {@link Error} which is set
+     * to non `null` value when scanning failed. You have to start scanning process again if that happens. Second argument
+     * is a scanned {@link Device}.
+     */
+    startDeviceScan(
+      UUIDs: UUID[] | null,
+      options: Partial<ScanOptions>,
```

Comment:
Partial is not needed after fixing ScanOptions: `options: ScanOptions | 'null',`